### PR TITLE
fix ServiceMeshServiceDispatcher overwrite ServiceDispatcher debug file when open SW_OAL_ENGINE_DEBUG.

### PR DESCRIPTION
**Pull request by zhyyu**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

The `org.apache.skywalking.oap.server.core.source.oal.rt.dispatcher.ServiceMeshServiceDispatcher` class will overwrite `ServiceDispatcher.class`. Change class file name logic when generate `Dispatcher` class in `openEngineDebug` circumstance.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Could you explain more about how this override happens? They are two classes, why would one will override the other?

File: oap-server/oal-rt/src/main/java/org/apache/skywalking/oal/rt/OALRuntime.java [hunk 1]
```diff
@@ -523,6 +524,9 @@ private void writeGeneratedFile(CtClass metricsClass, String className, String t
                 if (!folder.exists()) {
                     folder.mkdirs();
                 }
+                if (StringUtils.equalsIgnoreCase("dispatcher", type)) {
+                    className = metricsClass.getSimpleName();
+                }
```
Comment (kezhenxu94) [inline oap-server/oal-rt/src/main/java/org/apache/skywalking/oal/rt/OALRuntime.java:529]: This looks to be not an elegant solution, maybe we can just simplify this method's signature to `private void writeGeneratedFile(CtClass metricsClass, String type) throws OALCompileException` and use `metricsClass.getSimpleName()` to replace the original argument `className`?

**Comment by wu-sheng** [comment]
Be careful of full class name, it would be an issue for window(too long file path, sometimes show on git clone as well.)

**Comment by wu-sheng** [comment]
As this file is generated sequentially, you could generate `_index` when the name is in conflict. 
In your case, maybe `ServiceDispatcher.class` and `ServiceDispatcher_1.class` are a better choice. After all, this is not a runtime name.

**Comment by zhyyu** [comment]
> As this file is generated sequentially, you could generate `_index` when the name is in conflict. In your case, maybe `ServiceDispatcher.class` and `ServiceDispatcher_1.class` are a better choice. After all, this is not a runtime name.

In my case, in the file content, the class name is `ServiceMeshServiceDispatcher`, but the file name is `ServiceDispatcher.class`. I have used `metricsClass.getSimpleName() `to replace the original argument `className` to fix the problem.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**